content/en/docs/contribute/style/diagram-guide.md

```diff
@@ -0,0 +1,577 @@
+```
+Here are several items to consider when adding diagram captions: 
+
+* Use`<figcation></figcaption>` tags to add a diagram caption for `Inline`, `Mermaid+SVG` and `External Tool` diagrams. 
+* Wrap caption text inside `<p></p>` tags using `style="padding-left: 310px; font-weight: bold"`. This provides a consistent style for the caption text.
+* Prepend your diagram caption with `Figure NUMBER.`. You must use `Figure` and the number must be unique for each diagram in your documentation page. 
+* Add your diagram caption text after the `Figure NUMBER.` on the same line. Keep it short.
+* Position your diagram caption __BELOW__ your diagram.
+* You can hyperlink your caption using `<a></a>` tags.
+
+Here is the caption for the `/images/docs/components-of-kubernetes.svg` diagram: 
+```text
+<figcaption><p style="padding-left: 310px; font-weight: bold">Figure 4. Kubernetes Architecture.</p></figcaption><br>
+```
+Here is the hyperlinked caption for the same diagram:
+```text
+<figcaption><a href="https://kubernetes.io/docs/concepts/overview/components/"><p style="padding-left: 310px; font-weight: bold">Figure 4. Kubernetes Architecture Test.</p></a></figcaption><br>
+```
+
+Check out the source code for this page, `diagram-guide.md`, for more examples of diagram captions.
+
+**Diagram Referral**
+
+Finally, you can add a diagram caption referral. This is used inside your text. It allows a user to connect your text with the diagram. The `Figure NUMBER` in your referral and caption must match.   
+
+Here is an example of a caption referral included in your content text:
+```text
+Figure 9 depicts the components of the Kubernetes architecture. The control plane ...
+```
+Diagram referrals are optional and there are cases where they might not be suitable. If you are not sure, add a diagram referral to your text to see if it looks and sounds okay. When in doubt, use a diagram referral.
+
+**Complete picture**
+
+Figure 9 shows the Kubernetes Architecture diagram that includes the diagram, diagram caption and diagram caption referral. The `{{</* figure */>}}` shortcode renders the diagram. The `<figcaption>` defines the caption. The caption referral is contained in this paragraph.
+
+Here are the `{{</* figure */>}}` shortcode and `<figcaption>` statements for this diagram:
+```
+{{</* figure src="/images/docs/components-of-kubernetes.svg" alt="Kubernetes pod running inside a cluster" class="diagram-large" */>}}
+<figcaption><a href="https://kubernetes.io/docs/concepts/overview/components/"><p style="padding-left: 310px; font-weight: bold">Figure x. Kubernetes Architecture Test.</p></a></figcaption><br>
+```
+
+{{< figure src="/images/docs/components-of-kubernetes.svg" alt="Kubernetes pod running inside a cluster" class="diagram-large" >}}
+<figcaption><p style="padding-left: 310px; font-weight: bold">Figure 9. Kubernetes Architecture.</p></figcaption><br>
```

Comment:
It's not good practice to have a `<figcaption>` outside of the `<figure>` element it's for. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption

Reply:
Yes, aware of that. My thinking was:

- We don't have a consistent/suitable caption format for all diagram types. We need a shortcode for this.
- we want to avoid any more html than what is necessary, like <figure> tags. Thinking as someone who has never seen html/css. 
- <figcaption> works as "placeholder" for new "{{< caption >}} shortcode."

Suggest we keep as is, then open PR for new caption shortcode? Or revise text so that
```
<figure><figcaption> ... </figcaption></figure>
``` 
is used now and new caption shortcode PR follows. Guidance needed.

Reply:
Let's save that _discussion_ for the new issue / PR. I definitely recommend documenting `{{< figure >}}` shortcodes, with no extra HTML, for now.

BTW https://gohugo.io/content-management/shortcodes/#figure does already support a `caption` attribute.

Reply:
`{{< figure >}}` shortcode is in PR there for `.svg` cases. However it does not apply to the `Inline` method since there is no `.svg` image file.  This shortcode's caption attribute left justify's the caption text. This does not look right. 

Figure captions (Figure Number. Caption text) are essential for good technical documentation. IMHO, we need a uniform style and method right now for contributors to include figure captions for the `Inline`, `Mermaid+SVG` and `External Tool` methods. 

The only solution that I see right now is to tell contributors to cut/paste this code to insert captions:
```
<figure><figcaption><p style="padding-left: 310px; font-weight: bold">Figure #. CAPTION TEXT</p></figcaption></figure><br>
```
and of course update the Figure #. CAPTION TEXT portion.

Reply:
I firmly recommend saving the discussion on captioning for a follow up PR. That way, we can land the rest of the changes and then look into the best way to set up effective captioning, including how we align and style the text of the caption.

Reply:
To clear on what I want to _omit_: advice to readers about adding an invalid, standalone `<figcaption>`. If _this_ page uses standalone `<figcaption>` elements that's not great; we can accept it and plan to revisit.

But if we are telling everyone who reads the page to do the wrong thing, that takes much more effort to rework. I don't want to merge a page that gives advice that that I would immediately want to strike out.

Reply:
New contributor, diagram advocate hat on: :-)
- diagram captions are a must. Sporadically used through out docs. Figure shortcode is fine for `.svg` files. *We have nothing for Mermaid diagrams.*
- diagram referrals (e.g. Figure 9 illustrates ...) needed. Hard link between some text and actual figure. [Spatial](https://developers.google.com/style/images#figure-captions) approaches are buried in the text somewhere forcing users to jump around. Lots of text in the docs omit or use spatial approaches.
- I understand best practice use of <figure><figcaption> ... tags.
- We need single, uniform, consistent, documented method for adding figure captions.
New contributor, diagram advocate hat off: :-)

To progress this much needed PR: 
- Recommend `.svg` figure captions use figure shortcode caption attribute.
- Recommend `Inline` method Mermaid diagrams use text.

We can later on open PR to fix figure shortcode to handle Mermaid diagrams. How does that sound?

Reply:
SGTM @chrismetz09

Reply:
Okay got it! Tx for the lengthy discussion @sftim. Commits on the way.